### DISCUSSION TITLE
Feedback for damaged relay module or valve

### DISCUSSION THREAD

**Discussion by PlaneStuffRC** [Dan-in-CA/SIP#264] [Ideas]
Last week i had an issue with my system. Because the raspberry pi and the relay module were put in an area with high humidity, water got inside the box. It was a box rated with IP65, but some how water accumulated in the lower part of the box, where the 8ch relay module was. (maybe failed gasket). 
At that point i was trying to understand what happened, why the valves are not working, from the SIP the command was sent but no action on the valve. Then i opened to box, later to discover the relay board was destroyed.
Now, I'm thinking that develop a module that can provide feedback to SIP and signal if any valve are failing.

[MC33879 ](https://www.nxp.com/products/power-management/smart-switches-and-drivers/configurable-switches/configurable-octal-serial-switch:MC33879) will be a good candidate because it can detect an open circuit and over-current condition.
At this point i can design the hardware and manufacturing it and also do the firmware for it.
Probably we can make it work with multiple communications protocols like: can-bus, mod-bus, Ethernet TCP, etc.

But i will need some volunteers to help me with the SIP plugins :)

**Comment by mattip** [comment]
Since the proof of work is water being delivered to the plants, I think a better strategy would be to add a flow meter to your system, along the lines of issue #234 which I never finished updating

**Reply by Dan-in-CA** [reply]
Adrian,
This sounds very interesting.
I would be interested in working on the plugin part as time permits.

Dan
